icu4c/source/i18n/gregoimp.cpp

```diff
@@ -130,17 +130,15 @@ void Grego::dayToFields(double day, int32_t& year, int32_t& month,
     UBool isLeap = isLeapYear(year);
     
     // Gregorian day zero is a Monday.
-    dow = (int32_t) uprv_fmod(day + 1, 7);
-    dow += (dow < 0) ? (UCAL_SUNDAY + 7) : UCAL_SUNDAY;
+    dow = ((int32_t) day + 1) % 7;
```

Comment:
You changed the signature of `julianDayToDayOfWeek()` to reflect the fact that everybody was passing in an `int32_t` anyway, and it simplified the code inside more-- you didn't need all the casts.  It seems like you could do the same thing here.  I didn't do a comprehensive search of all the call sites, but I think I only found one that was using a `double`, and you could do the cast there.  Or am I missing something?

Reply:
I was under the impression that `Grego` was a public API so I refrained from changing the declaration, and there are also a couple other functions in there taking a double which should probably be changed as well. It seems it is not actually a public API so I'll change this too.